src/etc/inc/certs.inc

```diff
@@ -601,15 +614,27 @@ function cert_get_issuer($str_crt, $decode = true) {
 }
 
 /* Works for both RSA and ECC (crt) and key (prv) */
-function cert_get_publickey($str_crt, $decode = true, $type = "crt") {
+function cert_get_publickey($str_crt, $decode = true, $type = "crt", $password = "") {
 	if ($decode) {
 		$str_crt = base64_decode($str_crt);
 	}
 	$certfn = tempnam('/tmp', 'CGPK');
 	file_put_contents($certfn, $str_crt);
 	switch ($type) {
 		case 'prv':
-			exec("/usr/bin/openssl pkey -in {$certfn} -pubout", $out);
+			if ($password) {
+				$passfn = tempnam('/tmp', 'CGPK');
+				file_put_contents($passfn, $password);
+
+				exec("/usr/bin/openssl pkey -in {$certfn} -passin file:{$passfn} -pubout 2>/dev/null", $out, $status);
+
+				unlink($passfn);
```

Comment:
It's safer to use `unlink_if_exists()` here instead of `unlink()`. Will be the same for other uses throughout this code.

Reply:
Should I go ahead and change all uses of `unlink`?